### ISSUE TITLE
Better URL encoding

### ISSUE THREAD

**Issue by ad-si** [mochreach/midi-surf#22]
I didn’t check how exactly you create your URLs but it didn’t look like the Jsurl2 one, which seems to be the gold standard: https://github.com/wmertens/jsurl2

Love this project! Keep up the great work 😁

**Comment by ChrisWellsWood** [comment]
Hi @ad-si, glad you like the project! I took a quick look at jsurl2 and it looks great for sending JSON via URLs, but it looks like there's no compression, so it's going to be difficult to fit all the data for the controller into a URL without it being very long. The method I use requires JSON encoding the controller data structure to a string, compressing it with GZIP and then the bytes are base 64 encoded, all in the browser. The relevant Elm code can be found [here](https://github.com/mochreach/midi-surf/blob/5fbf45d45bf1fff7522e4e65d6fe7a97a762146a/src/Main.elm#L2855).

For example, for controller can be represented in JSON in around ~35,000 characters:
<img width="955" alt="image" src="https://github.com/mochreach/midi-surf/assets/8362681/7abed813-bc0b-4be8-a171-f0fb1d8bdc56">

If I convert to JSON, GZIP, then base64 encode, it reduces it down to a url containing ~2,250 characters, around a 15 times compression and the big advantage is that *all the data that describes the controller is in the URL, no server required!* I wrote a wee post about it on the Elm discourse forum, if you're interested: https://discourse.elm-lang.org/t/encoding-user-created-content-in-a-url/8914.
